app/src/main/java/dev/arkbuilders/arkmemo/graphics/SVG.kt

```diff
@@ -0,0 +1,168 @@
+        if (commandsArray.isNotEmpty()) {
+            if (paths.isNotEmpty()) paths.clear()
+            var path = AndroidDrawPath()
+            commandsArray.forEach { command ->
+                when (command.first().toString()) {
+                    MOVE_TO -> {
+                        path = AndroidDrawPath()
+                        val point = command.removePrefix("$MOVE_TO$SPACE")
+                        val pointList = point.split(SPACE)
+                        val x = pointList[0].toFloat()
+                        val y = pointList[1].toFloat()
+                        Log.d("svg", "move to $point")
+                        path.moveTo(x, y)
+                    }
+
+                    ABS_QUAD_TO -> {
+                        val points = command.removePrefix("$ABS_QUAD_TO$SPACE")
+                        val pointList = points.split(SPACE)
+                        val x1 = pointList[0].toFloat()
+                        val y1 = pointList[1].toFloat()
+                        val x2 = pointList[2].toFloat()
+                        val y2 = pointList[3].toFloat()
+                        Log.d("svg", "quad to $x1 $y1 $x2 $y2")
+                        path.quadTo(x1, y1, x2, y2)
+                    }
+
+                    ABS_LINE_TO -> {}
+                }
+                paths.add(DrawPath(path, paint))
+            }
+        }
+    }
+
+    companion object {
+        fun parse(path: Path) = SVG().apply {
+            val xmlParser = Xml.newPullParser()
+            var pathData = ""
+            xmlParser.apply {
+                setInput(path.reader())
+                var event = xmlParser.eventType
+                while (event != XmlPullParser.END_DOCUMENT) {
+                    val tag = xmlParser.name
+                    when (event) {
+                        XmlPullParser.START_TAG -> {
+                            when (tag) {
+                                SVG_TAG -> { viewBox = getAttributeValue("", Attributes.VIEW_BOX) }
+                                PATH_TAG -> {
+                                    strokeColor = getAttributeValue("", Attributes.Path.STROKE)
+                                    fill = getAttributeValue("", Attributes.Path.FILL)
+                                    pathData = getAttributeValue("", Attributes.Path.DATA)
+                                }
+                            }
+                        }
+                    }
+                    event = next()
+                }
+                createPaths(pathData)
+            }
+        }
+
+        private object Attributes {
+            const val VIEW_BOX = "viewBox"
+            const val XML_NS_URI = "xmlns"
+
+            object Path {
+                const val STROKE = "stroke"
+                const val FILL = "fill"
+                const val DATA = "d"
+            }
+        }
+    }
+}
+
+private const val MOVE_TO = "M"
+private const val ABS_LINE_TO = "L"
+private const val ABS_QUAD_TO = "Q"
+
```

Comment:
Now, you can make it even better by refactoring constant values of `String` type as variants of a enum type.
https://www.baeldung.com/kotlin/enum

Something like this:
```
enum class SvgCommand {
    MoveTo, AbsLineTo, AbsQuadTo
}
```

Pros of this approach:
1. Any command value is always valid. Malformed commands cannot exist.
2. You can declare command's parameters explicitly. In fact, the commands are not constants but objects. `MoveTo` must accept parameter to specify where to move. Same with `AbsLineTo` and `AbsQuadTo`.

Reply:
Do you mean the enums should have the coordinates parameters too?